src/core/connection.c

```diff
@@ -1402,7 +1407,7 @@ QuicConnOnShutdownComplete(
         Event.SHUTDOWN_COMPLETE.PeerAcknowledgedShutdown =
             !Connection->State.ShutdownCompleteTimedOut;
         Event.SHUTDOWN_COMPLETE.AppCloseInProgress =
-            Connection->State.AppCloseInProgress;
+            Connection->State.HandleClosed;
```

Comment:
This is a slight behavior change. It looks like HandleClosed will always be set no matter if close is called from the worker thread or an app thread, which is what this variable was used to detect.

Reply:
Actually, double checking this might be OK.

Reply:
Yeah. I think it was actually a bug that it wasn't set when called on a callback thread before. The new behavior is the most correct, IMO.